### PR TITLE
examples: ensure mod and fn are pub so that doc page works

### LINKED ISSUE #529: Generated documentation doesn't show up on qobject::T
#518 has shown that both generated docs, as well as user written docs don't show up in the documentation.
This is likely a scoping issue in our `mod cxx_qt_ffi`, which confuses the doc generator.

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
This then shows how a doc page is generated correctly for our QObjects.

Closes #529

### CODE REVIEW

File: examples/qml_features/rust/src/invokables.rs [hunk 1]
```diff
@@ -64,22 +64,22 @@ impl Default for RustInvokablesRust {
 // ANCHOR: book_invokable_impl
 impl qobject::RustInvokables {
     /// Immutable invokable method that returns the QColor
-    fn load_color(&self) -> Result<QColor, i32> {
+    pub fn load_color(&self) -> Result<QColor, i32> {
         Ok(self.as_qcolor())
```
Comment (LeonMatthesKDAB) [inline examples/qml_features/rust/src/invokables.rs:68]: Maybe we want to make `as_qcolor` pub as well?
Just to show how documentation looks like if you also have methods on the Rust struct.

File: examples/qml_features/rust/src/nested_qobjects.rs [hunk 2]
```diff
@@ -72,10 +69,11 @@ impl Default for OuterObjectRust {
...
-    //
-    // This method needs to be unsafe otherwise clippy complains that the
-    // public method might dereference the raw pointer.
-    unsafe fn print_count(self: Pin<&mut Self>, inner: *mut qobject::InnerObject) {
+    ///
+    /// # Safety
+    ///
+    /// This methods take a pointer so should consider safety when using the pointer
+    pub unsafe fn print_count(self: Pin<&mut Self>, inner: *mut qobject::InnerObject) {
         if let Some(inner) = unsafe { inner.as_ref() } {
```
Comment (LeonMatthesKDAB) [inline examples/qml_features/rust/src/nested_qobjects.rs:77]: This `unsafe` block can go now, as the whole method is marked `unsafe`.

Comment (LeonMatthesKDAB) [reply examples/qml_features/rust/src/nested_qobjects.rs:77]: Same for the `self.called` below.

**Review by LeonMatthesKDAB — COMMENTED** [review]
Mostly looks good, just a few nitpicks

File: examples/qml_features/rust/src/custom_parent_class.rs [hunk 3]
```diff
@@ -81,7 +81,7 @@ pub struct CustomParentClassRust {
 
 impl qobject::CustomParentClass {
     /// Override QQuickPaintedItem::paint to draw two rectangles in Rust using QPainter
-    fn paint(self: Pin<&mut Self>, painter: *mut qobject::QPainter) {
+    pub fn paint(self: Pin<&mut Self>, painter: *mut qobject::QPainter) {
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply examples/qml_features/rust/src/custom_parent_class.rs:84]: done

File: examples/qml_features/rust/src/invokables.rs (hunk 1 above)
Comment (ahayzen-kdab) [reply examples/qml_features/rust/src/invokables.rs:68]: done

File: examples/qml_features/rust/src/nested_qobjects.rs (hunk 2 above)
Comment (ahayzen-kdab) [reply examples/qml_features/rust/src/nested_qobjects.rs:77]: done

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM

**Pull request merged**